database/insertData/_helpers/core_mutations.js

```diff
@@ -5,10 +5,43 @@ GraphQL Fragment - CORE ACTIONS
 exports.coreActions = `
     # ON_APPLICATION_CREATE
     # change status to DRAFT
+    # generate serial
+    # generate initial name
     {
-        actionCode: "incrementStage"
+        actionCode: "generateTextString"
         sequence: 1
         trigger: ON_APPLICATION_CREATE
+        parameterQueries: {
+          pattern: "S-[A-Z]{3}-<+dddd>"
```

Comment:
Maybe good to have a comment here on the top explaining what this code means? Just briefly

Reply:
Sure, but I'll wait until we know what the actual pattern should be. I need to do overall documentation though, so will do now.

Reply:
Thanks. And regarding this TODO:
> Decide best way to store each template's serial pattern. I think the initial serial generation needs to be part of core_actions, since it's a fundamental requirement for front-end functioning. However, if we want each template type to have its own distinct serial number pattern, we have to store it somewhere specific to each template. I vote for a new "serial_pattern" field on "template", which would be part of applicationData so it could be inserted by this core action.

I think that's a good way to store the pattern in the template, that's nice because when we duplicate it also will have the pattern persisted in the previous template version, which can be updated if the user would like to. Are you willing to do this change as part of this PR?

Reply:
> I think that's a good way to store the pattern in the template, that's nice because when we duplicate it also will have the pattern persisted in the previous template version, which can be updated if the user would like to. Are you willing to do this change as part of this PR?

Yes, for sure. I want to, but Andrei had some reservations about it, so we should probably have a quick discussion about it as a group first.

Reply:
I've made a PR based off this one that implements this in a fairly straightforward way. See what you think: #459